### PR TITLE
location-verification.yaml v0.2 Release Candidate

### LINKED ISSUE #102: Documentation updates
**Problem description**
Two issues:
- `lastLocationTime` added in `VerifyLocationResponse` schema needs to be reflected in the examples.
- When `maxAge` conditions fail, it can result in `UNKNOWN` location response as discussed in [here](https://github.com/camaraproject/DeviceLocation/issues/34). However, it is better to capture the conclusion in the specification

**Expected action**
Spec update

### PR DESCRIPTION
**Pull request by jlurien**
#### What type of PR is this?

* correction
* cleanup
* documentation

#### What this PR does / why we need it:

- Update documention in info.description with latest atributes included
- Examples updated and enhanced
- Document behaviour when maxAge cannot be satisfied
- Removed required for `lastLocationTime` in response as there may be cases when there is no historical location data for the device
- IPV6 address must identify a single address  

#### Which issue(s) this PR fixes:

Fixes #102 , #112

### CODE REVIEW

**Review by sfnuser — COMMENTED** [review]
Thanks for the updates. It is clear now.
I still have issues with Golang generated code when dealing with polymorphism. But that's a codegen issue as you mentioned. I will modify the code accordingly.

**Review by bigludo7 — DISMISSED** [review]
LGTM

File: code/API_definitions/location-verification.yaml
Comment (jlurien) [inline code/API_definitions/location-verification.yaml:14]:
```suggestion
    * When the requested area does not match the area where the network locates the device. the verification result is `FALSE` . 
```

File: code/API_definitions/location-verification.yaml
Comment (rartych) [inline code/API_definitions/location-verification.yaml:66]: Current discussions in Commonalities tends to make `tags` optional and style it with "Title Case" 

```suggestion
        - Location Verification
```

In such simple API tags are not very much needed. If we keep it, then the global tags component with description is needed like this:

```
tags:
  - name: Location Verification
    description: Management of device location verification
```

**Comment by alpaycetin74** [comment]
Is it helpful to extend the definition of UNKNOWN, to state it also covers the case where there is a location, but older than maxAge ?
It could read: "when the network cannot locate the device within the specified maxAge, or cannot locate at all"
image

File: code/API_definitions/location-verification.yaml
Comment (alpaycetin74) [inline code/API_definitions/location-verification.yaml:13]: If the network's estimation of the device's location is fully contained within the requested area, the verification result is `TRUE`

Comment (jlurien) [reply code/API_definitions/location-verification.yaml:13]: Thanks @alpaycetin74, works for me

File: code/API_definitions/location-verification.yaml
Comment (alpaycetin74) [inline code/API_definitions/location-verification.yaml:14]: I'd recommend:
If the network's estimation of the device's location does not overlap with the requested area at all, the verification result is 'FALSE'

File: code/API_definitions/location-verification.yaml
Comment (jlurien) [inline code/API_definitions/location-verification.yaml:13]:
```suggestion
    * If the network's estimation of the device's location is fully contained within the requested area, the verification result is `TRUE`.
```

File: code/API_definitions/location-verification.yaml
Comment (jlurien) [inline code/API_definitions/location-verification.yaml:14]:
```suggestion
    * If the network's estimation of the device's location does not overlap with the requested area at all, the verification result is `FALSE` . 
```

File: code/API_definitions/location-verification.yaml
Comment (alpaycetin74) [inline code/API_definitions/location-verification.yaml:15]: I'd recommend:
If the network's estimation of the device's location partially overlaps with the requested area, or it is fully contained within the requested area, the result is 'PARTIAL'.  In this case, a `match_rate` may be included in the response, indicating an estimation of the likelihood of the match in percent.

**Comment by alpaycetin74** [comment]
Apologies, I think I got the PARTIAL one wrong. It should read:
If the network's estimation of the device's location partially overlaps with the requested area, or it fully contains the requested area, the result is 'PARTIAL'.
Thank you

File: code/API_definitions/location-verification.yaml
Comment (jlurien) [inline code/API_definitions/location-verification.yaml:15]:
```suggestion
    * If the network's estimation of the device's location partially overlaps with the requested area, or it fully contains the requested area (because is larger), the result is 'PARTIAL'. In this case, a `match_rate` may be included in the response, indicating an estimation of the likelihood of the match in percent.
```

**Comment by jlurien** [comment]
> Apologies, I think I got the PARTIAL one wrong. It should read: If the network's estimation of the device's location partially overlaps with the requested area, or it fully contains the requested area, the result is 'PARTIAL'. Thank you

Corrected, no problem, I should have noticed

File: code/API_definitions/location-verification.yaml
Comment (alpaycetin74) [inline code/API_definitions/location-verification.yaml:344]: Estimation of the match rate between Requested Area R and Network Area N, calculated as the percent value of Area(R ∩ N) / Area(N). Included only if VerificationResult is PARTIAL.

Comment (bigludo7) [reply code/API_definitions/location-verification.yaml:344]: Yes agree -- probably better to be very explicit. 
description: Match rate estimation for the location verification in percent - _may be valued only for PARTIAL result_.

File: code/API_definitions/location-verification.yaml
Comment (jlurien) [inline code/API_definitions/location-verification.yaml:15]:
```suggestion
    * If the network's estimation of the device's location partially overlaps with the requested area, or it fully contains the requested area (because it is larger), the result is 'PARTIAL'. In this case, a `match_rate` may be included in the response, indicating an estimation of the likelihood of the match in percent.
```

File: code/API_definitions/location-verification.yaml
Comment (jlurien) [reply code/API_definitions/location-verification.yaml]: Rephrased

File: code/API_definitions/location-verification.yaml
Comment (jlurien) [inline code/API_definitions/location-verification.yaml:344]:
```suggestion
      description: Estimation of the match rate between the area in the request (R), and area where the network locates the device (N), calculated as the percent value of the intersection of both areas divided by the network area, that is (R ∩ N) / N * 100. Included only if VerificationResult is PARTIAL.
```

**Comment by bigludo7** [comment]
@jlurien We have still to wait for IP v6 fix before to merge this one - correct?

**Review by bigludo7 — COMMENTED** [review]
A small proposal for lastLocationTime description.

**Comment by jlurien** [comment]
> @jlurien We have still to wait for IP v6 fix before to merge this one - correct?

It is almost closed in QoD, so I hope we can apply it by tomorrow. We'll have to align the 3 specs

**Review by bigludo7 — APPROVED** [review]
LGTM

**Review by alpaycetin74 — APPROVED** [review]
LGTM, thank you

**Pull request merged**
